collections/soldiers.js

```diff
@@ -0,0 +1,47 @@
+const MongoClient = require('mongodb').MongoClient;
+const assert = require('assert');
+const configs = require('../config');
+const utils = require("../utility/utility");
+
+// Connection URL
+const url = configs.dbUrl;
+
+// Database Name
+const dbName = configs.dbName;
+
+// Use connect method to connect to the server
+function newSoldier(collection, data, errorHandler) {
+  if (data["id"] && !(utils.isMatchesRegex(/[0-9]{7}/, data["id"]))) {
+    errorHandler.notValidData("The soldier json that was sent didnt had a legal id");
+  } else {
+    let fields = ["id", "name", "rank", "limitations"];
+
+    for (let i = 0; i < fields.length; i++) {
+      if (!data[fields[i]]) {
+        errorHandler.notValidData("The soldier json that was sent is missing the field " + fields[i]);
```

Comment:
if this throws an error then the return is unreachable. if not then the error is not recieved